CMakeLists.txt

```diff
@@ -704,13 +716,19 @@ function(build_tests)
       configure_file(EXAMPLES/MATRIX_MARKET/B.mtx  ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/B.mtx)
       configure_file(EXAMPLES/MATRIX_MARKET/Bz.mtx ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/Bz.mtx)
       configure_file(EXAMPLES/MATRIX_MARKET/arpackmm.sh ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/arpackmm.sh)
-      add_test(NAME arpackmm_tst WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY} COMMAND ${BASH_PROGRAM} -eu arpackmm.sh)
+      add_test_with_working_dir(arpackmm_tst
+        WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
```

Comment:
Can you move the `WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}` and the `COMMAND` inside the `add_test_with_working_dir` function

Reply:
That would make that function less versatile.
Let's say a future test would require the working directory to be set to somewhere in the source tree. Then, the new function wouldn't work.

I also don't know why you mean by moving the `COMMAND` to the new function. It is different for each of the three tests.